### PR TITLE
Remove unnecessary calls to GetTypeInfo

### PR DESCRIPTION
**Pull request by eerhardt**
GetTypeInfo() calls are no longer necessary just to use Reflection APIs. And typically they [allocate a small `TypeDelegator` object](https://github.com/dotnet/runtime/blob/88ecbab85beead9c7d9406ee2f3ea468b1d4cd1e/src/libraries/System.Private.CoreLib/src/System/Reflection/IntrospectionExtensions.cs#L16) when used, so they should be avoided when not necessary.

I removed the obvious cases from the product, and a few occurrences in `test` code. However, I didn't tackle every place `test` code was calling it as there is a diminishing return.

Similar work is done in dotnet/runtime:

* https://github.com/dotnet/runtime/pull/44414
* https://github.com/dotnet/runtime/pull/44891

### CODE REVIEW

File: src/Shared/ActivatorUtilities/ActivatorUtilities.cs [hunk 1]
```diff
@@ -47,13 +47,11 @@ public static object CreateInstance(IServiceProvider provider, Type instanceType
...
-            if (!instanceType.GetTypeInfo().IsAbstract)
+            if (!instanceType.IsAbstract)
             {
-                foreach (var constructor in instanceType
-                    .GetTypeInfo()
-                    .DeclaredConstructors)
+                foreach (var constructor in instanceType.GetConstructors())
                 {
-                    if (!constructor.IsStatic && constructor.IsPublic)
+                    if (!constructor.IsStatic)
```
Comment (eerhardt) [inline src/Shared/ActivatorUtilities/ActivatorUtilities.cs:54]: Note these changes match what was done in dotnet/runtimes ActivatorUtilities. See https://github.com/dotnet/runtime/pull/40227/files#diff-d04fd33a5eba20dc5b742e17b751d3b9c0860d114b73c777a0e140c1ceaccf6fR57

Comment (davidfowl) [reply src/Shared/ActivatorUtilities/ActivatorUtilities.cs:54]: Hmmmm when?

Comment (halter73) [reply src/Shared/ActivatorUtilities/ActivatorUtilities.cs:54]: Is `constructor.IsPublic` implied? Or does this change ActivatorUtilities.CreateInstance to use non-public ctors? If it's the latter, this seems like a fairly major breaking change.

**Comment by Pilchie** [comment]
Throwing a bunch of areas on for people to take a look.

File: src/Shared/ActivatorUtilities/ActivatorUtilities.cs (hunk 1 above)
Comment (BrennanConroy) [reply src/Shared/ActivatorUtilities/ActivatorUtilities.cs:54]: Previous conversation about this: https://github.com/dotnet/runtime/pull/40227/files#r463897549
TLDR: `GetConstructors` returns only public constructors

Comment (eerhardt) [reply src/Shared/ActivatorUtilities/ActivatorUtilities.cs:54]: Just looked again, it also only returns `Instance` constructors, so we could remove the `IsStatic` check from both copies (runtime and aspnetcore) of this code.

https://github.com/dotnet/runtime/blob/1821d9c14b970d58e0768256de138b6c0287e07d/src/libraries/System.Private.CoreLib/src/System/Type.cs#L175-L178

**Review by Tratcher — APPROVED** [review]

**Review by halter73 — APPROVED** [review]
